target/linux/mediatek/dts/mt7622-unielec-u7622-02.dts

```diff
@@ -0,0 +1,549 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR MIT)
+/*
+ * Copyright (c) 2017 MediaTek Inc.
+ * Author: Ming Huang <ming.huang@mediatek.com>
+ *	   Sean Wang <sean.wang@mediatek.com>
+ */
+
+/dts-v1/;
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+#include "mt7622.dtsi"
+#include "mt6380.dtsi"
+
+//#define USE_PCIE1
```

Comment:
Mhhh why? is there a variant of this device that use pcie1?

Reply:
> Mhhh why? is there a variant of this device that use pcie1?

U7622-02 use 4 capacitors to switch between PCIE1 and SATA

Reply:
So again what is the default state and how to transition from the 2 state?

Do you have to do hw modification to change this? GPIO? a hw switch on the board?

We can't have these kind of commented define and ifdef in dts. To support both of them you should make a dtsi and produce 2 different dts and 2 variant of this device.

Reply:
Customers define the default state. Unlike banana-pi-r64 using GPIO to switch between PCIE1 and SATA, U7622-02 use 4 capacitors to switch, so you have to to move the 4 capacitors to decide the signals connected to M.2 PCIE or SATA3.0.

Reply:
Alright, I will add a uboot env option "use_sata" to decide it. During booting kernel, a uboot script will change the kernel dts according to the use_sata uboot env value.

Reply:
@hashyuan ok it's not the best solution but it's a solution :D

I would ask you to add some comments in the dts about the related nodes 
and improve the documentation about use_sata

Something like
```
U7622-02 use 4 capacitors to switch, so you have to to move the 4 capacitors to decide the signals connected to M.2 PCIE or SATA3.0.
Uboot will patch the dts to enable and disable the related nodes.
To make the switch set use_sata in the uboot env........
```

This is just an example but it's really to not lose info about this process and the option after this gets merged. (on openwrt we check the commit but sometime we lose all the discussion done on the pr/mailing list so better put all the info in the commit description)

Reply:
A comment is added.